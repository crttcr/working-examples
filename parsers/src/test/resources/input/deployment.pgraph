// Example server resources
//
//
// NOTE: This deployment model points out the need for a reference capability in the model parser.
// It does not make sense to redefine elements that should be defined in a more narrow, appropriate scope.
//
// FIXME: Implement inclusion or reference capability in the parser
//
## vertex.auto.id.generator = STRING_SEQUENCE
## vertex.auto.id.prefix    = xiv.deployment

(s1: SERVER)
(s2: SERVER)
(s3: SERVER)
(b1: BROKER)
(b2: BROKER)
(t1: TOPIC)
(t2: TOPIC)
(t3: TOPIC)
(t4: TOPIC)
(v:  SERVICE)

(b1)--[:DEPLOYED_TO]->(s1)
(b2)--[:DEPLOYED_TO]->(s2)
(v )--[:DEPLOYED_TO]->(s3)

(v )--[:READS_FROM]->(t1)
(v )--[:WRITES_TO]->(t2)
(v )--[:WRITES_TO]->(t3)
(v )--[:WRITES_TO]->(t4)
