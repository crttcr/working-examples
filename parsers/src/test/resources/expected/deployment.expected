## vertex.auto.id.generator = STRING_SEQUENCE
## vertex.auto.id.prefix = xiv.deployment
VERTEX: s1
	LABEL: SERVER
VERTEX: s2
	LABEL: SERVER
VERTEX: s3
	LABEL: SERVER
VERTEX: b1
	LABEL: BROKER
VERTEX: b2
	LABEL: BROKER
VERTEX: t1
	LABEL: TOPIC
VERTEX: t2
	LABEL: TOPIC
VERTEX: t3
	LABEL: TOPIC
VERTEX: t4
	LABEL: TOPIC
VERTEX: v
	LABEL: SERVICE
EDGE: DEPLOYED_TO
EDGE: DEPLOYED_TO
EDGE: DEPLOYED_TO
EDGE: READS_FROM
EDGE: WRITES_TO
EDGE: WRITES_TO
EDGE: WRITES_TO